plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/messenger/ChatPromptHandler.kt

```diff
@@ -32,6 +32,15 @@ class ChatPromptHandler(private val telemetryHelper: TelemetryHelper) {
     private var requestId: String = ""
     private var statusCode: Int = 0
 
+    private fun countTotalNumberOfCodeBlocks(message: StringBuilder): Int {
```

Comment:
Same applies here: [Change calculation method to `mynahUI.endStream`](https://github.com/aws/aws-toolkit-vscode/pull/4761#discussion_r1574313719).

As we discussed several times that this method will not work properly especially in case there are code block markdowns inside code blocks. 

Please use `mynahUI.endMessageStream` instead:
Send message to webview from IDE, grab the latest count by ending the stream on presentation layer with mynahUI.endStream and send it back to IDE level from webview.